src/bootstrap/builder.rs

```diff
@@ -97,18 +97,36 @@ impl RunConfig<'_> {
         self.builder.build.build
     }
 
-    /// Return a `-p=x -p=y` string suitable for passing to a cargo invocation.
+    /// Return a list of crate names selected by `run.paths`.
     pub fn cargo_crates_in_set(&self) -> Interned<Vec<String>> {
         let mut crates = Vec::new();
         for krate in &self.paths {
             let path = krate.assert_single_path();
             let crate_name = self.builder.crate_paths[&path.path];
-            crates.push(format!("-p={crate_name}"));
+            crates.push(crate_name.to_string());
         }
         INTERNER.intern_list(crates)
     }
 }
 
+/// A description of the crates in this set, suitable for passing to `builder.info`.
+///
+/// `crates` should be generated by [`RunConfig::cargo_crates_in_set`].
+pub fn crate_description(crates: &[impl AsRef<str>]) -> String {
+    if crates.is_empty() {
+        return "".into();
+    }
+
+    let mut descr = String::from(" {");
+    descr.push_str(crates[0].as_ref());
+    for krate in &crates[1..] {
+        descr.push_str(", ");
+        descr.push_str(krate.as_ref());
+    }
+    descr.push('}');
```

Comment:
FWIW, this seems like it could be near-identically replicated with https://doc.rust-lang.org/nightly/std/fmt/struct.DebugSet.html, at the cost of `{"rustc_interface"}` rather than `{rustc_interface}`.

But :shrug: either way -- the implementation is pretty simple.

Reply:
all the `fmt::Debug*` types only work inside a `fmt::Debug` impl unfortunately, since we need access to a `fmt::Formatter` :disappointed: and at that point it doesn't make the code much shorter.